stdlib/public/Cxx/CxxSpan.swift

```diff
@@ -62,12 +80,15 @@ extension CxxSpan {
 @available(SwiftStdlib 6.1, *)
 extension Span {
   @_alwaysEmitIntoClient
-  @lifetime(immortal)
   @unsafe
+  @lifetime(borrow span)
```

Comment:
Unfortunately, I am not sure this would work for the purposes we want to use this conversion for. 

The planned usage looks like this. imagine the following C++ function:
```
using SpanOfInt = std::span<int>;
SpanOfInt foo(const Bar& b [[clang::lifetimebound]]);
```

We plan to import it like this:
```
@_disfavoredOverload
func foo(b: borrwing Bar) -> SpanOfInt 
```

And we plan to generate a safe wrapper using Swift's span something like this:
```
@lifetime(bar)
func foo(b: borrowing Bar) -> Span<int> {
  Span(_unsafeCxxSpan: foo(b))
}
```

Note that, we expect the Swift span to be used after the lifetime of the corresponding C++ span ends. I'd expect this to be OK as the C++ span does not actually own the buffer. So we want to inherit/copy the dependencies of the C++ span to the Swift span. Unfortunately, we cannot do that as the C++ span is escapable. This is why my initial feeling was to just make the Swift span immortal and leave it to the safe wrapper to propagate the correct lifetimes.

What do you think, what would be the proper way to make this scenario work?

Reply:
@Xazax-hun We have two options for ceating such wrappers:

1. Add `@_unsafeNonescapableResult` to the Span initializer:

```
@_unsafeNonescapableResult
Span.init<T: CxxSpan<Element>>(_unsafeCxxSpan span: borrowing T)`
```

This is ok if you never expect programmers to access this initializer. You're placing the responsibilty for safety entirely on the wrapper.

2. Add `_overrideLifetime` to the wrapper:

```
@lifetime(bar)
func foo(b: borrowing Bar) -> Span<int> {
  let cxxSpan = cxx_foo(b)
  let swiftSpan = Span(_unsafeCxxSpan: cxxSpan)
   // The lexical variable scope of cxxSpan covers the final use if `swiftSpan` below:
  return _overrideLifetime(swiftSpan, borrowing: b)
}
```